### DISCUSSION TITLE
Verb being dropped on the request

### DISCUSSION THREAD

**Discussion by afriedma** [dotnet/aspnetcore#26348] [General] [answered]
Hello,

In version Microsoft.AspNetCore.App 2.1.15, we are seeing a sporadic behavior, where you make to requests to a same endpoint, but its possible that a request can come with no method, like below
WebHost: Information: Request starting HTTP/1.1 http://localhost:35958/v1/bar

But on consecutive request, you will see 
WebHost: Information: Request starting HTTP/1.1 GET http://localhost:35958/v1/bar

We upgraded to 2.1.22 and the bug went away completely. But unfortunately our runtime environment doesn't support 2.1.22

Can anyone recall what changed between 2.1.15 and 2.1.22 that fixed this bug?  And can you help with workaround if such exist.

I apologize in advance for being a bit vague and happy to add more details if needed, but I am really hoping someone saw this behavior before and can respond right away. The reason I am saying it, is the method in the error message is blank, as well as, that endpoint does work sporadically.

Thank you in advance.

**Comment by Tratcher** [comment, marked as answer]
@halter73 this was one of your favorites 😉

**Reply by afriedma** [reply]
You guys seen this bug as well?  Can you point to a workaround?  It is driving me nuts!

**Reply by Tratcher** [reply]
Ah, found it in the archive: https://github.com/aspnet/KestrelHttpServer/issues/2591. In short, any code that tries to read the HttpRequest between active requests can cause it to cache invalid state. Loggers using IHttpContextAccessor are the most common cause.

**Reply by afriedma** [reply]
Hey, thanks for reply.  I did see that one, but we aren’t using any custom middleware.  It’s a simple Mvc set up.  With a bunch of AddScoped plus AddHttpClient configuration.  So it is using Factory, but I have to keep that code as is.

**Reply by Tratcher** [reply]
> But unfortunately our runtime environment doesn't support 2.1.22

You could always publish 2.1.22 with your app using a standalone deployment.
https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained

**Reply by afriedma** [reply]
That’s not possible, app is hosted by pivotal with their buildpack🙃

**Reply by afriedma** [reply]
await keyword was missing in one of the function calls, in case anyone comes across something similar.
